### ISSUE TITLE
Some questions about the dataset

### ISSUE THREAD

**Issue by crocodilegogogo** [uea-machine-learning/tsml_repo#39]
Hi,
    I am a researcher in machine learning, resently I was working on UCR archive for TSC study. I find recent techniques training epoch by epoch (e.g. deep learning) is hard to apply to UCR due to lacking training data. More seriously, there is no validation sets provided, and test sets is usually not allowed for model selection as mentioned in the article 'The UCR Time Series Archive'.
    I am very positive about the contribution of UCR to the field of TSC, but I think there may be some problems with the UCR dataset.
    Firstly, insufficient data. In the age of big data, I do not think the acquisition of data becomes the main obstacle for time series classification. However, lots of datasets in UCR is very small, especially training sets. And these data are at least, not impossible to acquire. So there's a conflict between UCR and the real world.
    Secondly, super parameters selection. Not all algorithms are suitable for cross-validation to select parameters or models (e.g. deep learning) due to computation cost. Validation sets are necessary for many machine learning algorithms. However, it is hard to split validation sets for small datasets. If we treat this problem from the perspective of real-world application, we should use test sets as validation sets. We train a model through training sets, and find the best model for test sets, then we apply this model to the real-world application. While most deep learning algorithms choose models with  minimum training loss, this apparently exists overfitting problems. An overfited model is useless for real-world application due to low generalization performance.
    I therefore recommend that existing test sets should be treated as validation sets. Actually this is pretty common in Computer Vision area. And the sizes of new datasets should be considered before they are included into UCR archive. Big data is more in line with the real world.  And some of small datasets should be removed. Otherwise it may hinder the development of new technology in time series classification, considering small sample study is not the original intention of UCR archive.
    Hope for your reply!

**Comment by TonyBagnall** [comment]
hi there, thanks for the input. I would disagree, the acquisition of labelled data is absolutely the main obstacle to many many applications I look at. For example, consider any medical application. The cost of a medical trial often means we only have tens or hundreds of cases in a data set. If an algorithm is not suited to small data sets, then that needs to be clearly acknowledged. We have 128 datasets in the archive, so selecting larger data based on the criteria that the algorithm you want to test is only good for larger data seems fine to me.

Have you read this paper on deep learning for TSC? 
https://link.springer.com/article/10.1007/s10618-019-00619-1

That being said, this issue is often raised, especially by deep learners. The train set must never be used in model fitting/selection, and I think it up to the algorithm designer to decide on how to use the train data, to me a fixed validation set makes no sense.  If you have access to large data sets for TSC we will happily add them to the archive. I have a few to go in this year when I get the time.

**Comment by crocodilegogogo** [comment]
Dear Professor Bagnall,
    Thank you for your patient reply. 
    I read the paper you mentioned before. Actually I don't approve the method used in that paper for model selection. Because minimum training loss apparently leads to serious overfitting. These overfitted models are useless for real world applications, and they are only effective on the UCR training sets. Now let's imagine a situation, we use the data of UCR to solve corresponding problem of the real world. How do we choose a model? If we don't split the training and test datasets again, the best way is, training datasets used for training models and test datasets for choosing models. Minimum training loss for model selection is apparently not suitable. From this point of view, using existing test datasets as validation sets, is at least not without research value. Much less existing DL algorithms are far away from reaching 100% accs even using test sets as validation sets.
    As far as I know, a standard machine learning data sets should include training sets, validation sets and test sets. The  training sets and validation sets are labelled, while test sets are unlabelled, we choose a model through training sets and validation sets, then test the test sets, finally upload the testing results to the dataset websites for obtaining scores. Of course some CV domains cannot provide validation sets due to lack of labelled data. These domains allow using test datasets as validation sets (e.g. domins of super-resolution, hyperspectral images whose datasets also only provide training and test sets). It is almost a consensus, to facilitate migrating works on datasets to real world applications.
    I strongly believe DL is very fit for TSC, the problem is not lack of training data, but model selection due to lack of validation sets. Existing articles using minimum training loss are obviously lacking of generalization ability. If two DL models is good enough, their differences vanish or hard to distinguish due to overfitting, through one model is apparently better. This leads the researchers to conclude that DL is not suitable for TSC, and with few research value. However, it is the constraints of UCR data sets leading to this phenomenon. This is the main reason why there are not many deep learning jobs based on UCR. Actually, I believe there are still lots of interesting work for DL to explore on UCR.
    So dear professor Bagnall, please access the test set as the validation set,  this will make the UCR dataset even more influential!

**Comment by ErnstHowie** [comment]
Hi @TonyBagnall . I have the same issue as @crocodilegogogo. Please give me some help. Thanks a lot!

**Comment by crocodilegogogo** [comment]
Dear Professor Bagnall,
     I have been thinking about your reply recently, and still get some confusions. If the test set is totally 'unseen', can ablation experiments be conducted on the test set?
     Hope for your reply!

**Comment by TonyBagnall** [comment]
hi, ablation studies are for finding out what component of an algorithm leads to its performance. This is an internal assessment of an algorithms relative performance. In this case, change of performance on test data is fine, If, however, the study is comparing algorithms, then test data must be completely unseen. At an extreme, a nearest neighbour classifier is always best if you can see the test data!

**Comment by crocodilegogogo** [comment]
I understand, thanks!
